index.js

```diff
@@ -1,21 +1,593 @@
+
+    for (const testRun of testRuns) {
+      //modify xctestrun with Scope variables
+
+      let plutilExportCommand =
+        "plutil -convert json -o " + testrunJson + ` "${testRun}"`;
+      await exec.exec(plutilExportCommand, null, null);
+
+      let jsonString = fs.readFileSync(testrunJson, "utf8");
+      const testTargets = JSON.parse(jsonString);
+
+      for (const target of Object.keys(testTargets)) {
+        if (target.charAt(0) !== "_") {
+          if (testTargets[target].TestingEnvironmentVariables) {
+            await insertEnvVariables(testRun, target);
+          } else if (target === "TestConfigurations") {
+            let configurationNumber = 0;
+            for (const configuration of testTargets["TestConfigurations"]) {
+              let testNumber = 0;
+              for (const test of configuration["TestTargets"]) {
+                await insertEnvVariables(
+                  testRun,
+                  target +
+                    "." +
+                    configurationNumber +
+                    "." +
+                    "TestTargets" +
+                    "." +
+                    testNumber
+                );
+              }
+            }
+          }
+        }
+      }
+      //run tests
+      let testCommand =
+        "xcodebuild test-without-building " +
+        codeCoverParam +
+        " -xctestrun " +
+        `"${testRun}"` +
+        ' -destination "' +
+        destination +
+        '"' +
+        extraParameters;
+      try {
+        await exec.exec(testCommand, null, null);
+      } catch (error) {
+        testError = error.message;
+      }
+    }
+    if (testError) {
+      core.setFailed(testError.message);
+      console.log(testError.message);
+    }
 
-    core.setOutput('time', new Date().toTimeString());
   } catch (error) {
     core.setFailed(error.message);
+    console.log(error.message);
+
+  }
+
+    //Clean up
+  fs.rmSync(sdkTestingDir, { recursive: true });
+
+}
+
+function getSDKForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "macosx";
+    case "tvos":
+      return "appletvsimulator";
+    default:
+      return "iphonesimulator";
+  }
+}
+
+function getDestinationForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "platform=macOS,arch=x86_64";
+    case "tvos":
+      return "platform=tvOS Simulator,name=Apple TV 4K";
+    default:
+      return "platform=iOS Simulator,name=iPhone 13";
+  }
+}
+
+function getFrameworkPathForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "macos-arm64_x86_64 ";
+    case "tvos":
+      return "tvos-arm64_x86_64-simulator";
+    default:
+      return "ios-arm64_x86_64-simulator";
+  }
+}
+
+async function deleteLinesContaining(file, match) {
+  let newName = file + "_old";
+  await io.mv(file, newName);
+  fs.readFile(newName, { encoding: "utf-8" }, function(err, data) {
```

Comment:
the callback for `fs.readFile` is not waited for nor handled so it can potentially leak. I'd either use `fs.readFileSync` (sync version) or do something like: 
```javascript
return new Promise((resolve, reject) => {
  fs.readFile(newName, { encoding: "utf-8" }, function(err, data) {
    // everything you have already
    fs.writeFile(file, updatedData, err => {
      if (err) {
        reject(err)
      }
      console.log("Successfully updated the file data");
      resolve()
    });
  })
})
```

Reply:
Changed to readFileSync